app/components/form.tsx

```diff
@@ -292,6 +297,90 @@ export const Input = ({
   </Box>
 );
 
+export const DayPickerField = ({
+  label,
+  name,
+  value,
+  onChange,
+  disabled,
+  controls,
+  ...props
+}: {
+  name: string;
+  value: Date;
+  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
+  controls?: ReactNode;
+  label?: string | ReactNode;
+  disabled?: boolean;
+} & Omit<DayPickerInputProps, "onChange">) => {
+  const handleDayClick = useCallback(
+    (day: Date) => {
+      const ev = {
+        currentTarget: {
+          value: day.toISOString().slice(0, 10),
+        },
+      } as ChangeEvent<HTMLSelectElement>;
+      onChange(ev);
+    },
+    [onChange]
+  );
+  const formatDateAuto = useTimeFormatUnit();
+  const inputProps = useMemo(() => {
+    return {
+      name,
+      formatDate: formatDateAuto,
+      value: formatDateAuto(value, TimeUnit.Day),
+      disabled,
+      style: {
+        padding: "0.625rem 0.75rem",
+        color: disabled ? "monochrome300" : "monochrome700",
+        fontSize: "1rem",
+        minHeight: "1.5rem",
+        display: "block",
+        borderRadius: "0.25rem",
+        width: "100%",
+        border: "1px solid var(--theme-ui-colors-monochrome500)",
+
+        // @ts-ignore
+        ...props.inputProps?.style,
+      },
+      ...props.inputProps,
```

Comment:
Not sure if `style` wouldn't be overwritten if `props.inputProps` contains them?

Reply:
Oups, you're right ;) Moved inputprops above.